larq_zoo/binary_alex_net.py

```diff
@@ -0,0 +1,174 @@
+from typing import Tuple
+
+import larq as lq
+import tensorflow as tf
+from zookeeper import ComponentField, Field, factory, task
+
+from larq_zoo import utils
+from larq_zoo.model_factory import ModelFactory
+from larq_zoo.train import TrainLarqZooModel
+
+
+@factory
+class BinaryAlexNetFactory(ModelFactory):
+    """
+    Implementation of ["Binarized Neural Networks"](https://papers.nips.cc/paper/6573-binarized-neural-networks)
+    by Hubara et al., NIPS, 2016.
+    """
+
+    inflation_ratio: int = Field(1)
+
+    input_quantizer = Field("ste_sign")
+    kernel_quantizer = Field("ste_sign")
+    kernel_constraint = Field("weight_clip")
+
+    @Field
+    def num_classes(self) -> int:
+        if self.dataset is None:
+            raise TypeError("Must override either `dataset` or `num_classes`.")
+        return self.dataset.num_classes
```

Comment:
Isn't this already inherited from the ModelFactory class?

Reply:
Thanks for spotting that!

Reply:
Removed in a4ce49b